### PR TITLE
MySQL allows to pass named parameters as object

### PR DESCRIPTION
**Pull request by gre000**
Hello,
only a small extension that allows to pass named parameters as object.
Example:
INSERT INTO database (`id`, `field1`) VALUES (:id, :field1)
Based on the documentation from mysql nodejs modul!
(https://www.npmjs.com/package/mysql#preparing-queries)

This is my first pull request, I hope it's ok. 

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by dceejay** [comment]
Hi, in general we prefer a discussion via the discourse forum first so we can get wider input to the new feature, but it seems ok to me. Let me check it out some more first but should be OK. thanks

File: storage/mysql/README.md [hunk 1]
```diff
@@ -27,3 +27,26 @@ If nothing is found for the key then <i>null</i> is returned.
...
+msg.payload=[24, 'Helloworld'];
+msg.topic="INSERT INTO database (`id`, `field1`) VALUES (?, ?);"
+return msg;
+```
+
+with named parameters:
+
+```javascript
+msg.payload={}
+msg.payload.id=42;
```
Comment (dceejay) [inline storage/mysql/README.md]: Example may be clearer if you used myid (or something other than id) instead so you could see which were the mysql variables and which were the javascript ones

File: storage/mysql/68-mysql.js [hunk 2]
```diff
@@ -122,7 +122,24 @@ module.exports = function(RED) {
...
+                        var bind = [];
+                        if(Array.isArray(msg.payload)){bind = msg.payload}
+                        else if(typeof msg.payload === 'object' && msg.payload !== null){
+                            bind=msg.payload;
+                            node.mydbConfig.connection.config.queryFormat = function (query, values) {
+                                if (!values){
+                                    return query;
+                                }
+                                return query.replace(/\:(\w+)/g, function (txt, key) {
+                                  if (values.hasOwnProperty(key)) {
```
Comment (dceejay) [inline storage/mysql/68-mysql.js]: we like consistent indents of 4 spaces :-)

File: storage/mysql/68-mysql.js [hunk 3]
```diff
@@ -122,7 +122,24 @@ module.exports = function(RED) {
                 if (node.mydbConfig.connected) {
                     if (typeof msg.topic === 'string') {
                         //console.log("query:",msg.topic);
-                        var bind = Array.isArray(msg.payload) ? msg.payload : [];
+                        var bind = [];
+                        if(Array.isArray(msg.payload)){bind = msg.payload}
```
Comment (dceejay) [inline storage/mysql/68-mysql.js]: we prefer spaces `if (`  and `) { ` for readability

File: storage/mysql/README.md [hunk 4]
```diff
@@ -27,3 +27,26 @@ If nothing is found for the key then <i>null</i> is returned.
...
+msg.payload.field1="Hello World";
+msg.topic="INSERT INTO database (`id`, `field1`) VALUES (:id, :field1) ON DUPLICATE KEY UPDATE `id`=:id,`field1`=:field1;"
```
Comment (dceejay) [inline storage/mysql/README.md]: as per above would be clearer if easy to tell which were the javascript ones and which were mysql ones

File: storage/mysql/68-mysql.js (hunk 2 above)
Comment (gre000) [reply storage/mysql/68-mysql.js]: I've changed

File: storage/mysql/68-mysql.js (hunk 3 above)
Comment (gre000) [reply storage/mysql/68-mysql.js]: I've changed

File: storage/mysql/README.md (hunk 4 above)
Comment (gre000) [reply storage/mysql/README.md]: I've changed

**Comment by dceejay** [comment]
Thanks - sorry it's taken until "lockdown" to get round to this.

**Pull request merged**